Adding support for stripWWW and stripFragments

hlysig opened the pull request:
With this pull requests users of this library can pass in option object
and utilize strip www and strip fragments from the normalize-url
library.

This is done by allowing users to pass in optional option parameter as
follows

    getUrls(text, {stripFragment: false});

and

    getUrls(text, {stripWWW: false});

This change is backwards compatible, where the defaults favor the
functionality as before.

Note that The normalize-url library was updated to 1.3.0.

sindresorhus commented:
Why? What use-case does this solve? I don't see the point of the `stripWWW` option here.

hlysig commented:
I am using this library to identify links in texts and map them into html anchors. The use-case this pull requests solves, with these modifications, is that the links written in the text are unchanged.

For example. If I use the following text

	"You can read my blog at http://www.simnet.is/~hlysig"

and I would like to replace the url with html anchor, it would be hard to replace the resulting url if it is modified. Also, one can not assume that www.domain.com and domain.com point to the same ip address. Thus the resulting url might be invalid.

I wrote the above mentioned url-anchor library today and I will publish it to Git tomorrow. It currently deepens on my fork of your project. If you don't see any value in this pull-request, you can delete it and I will stick to the fork.

hlysig commented:
This exposes the features in the normalize-url library that you accepted earlier today.

sindresorhus commented:
> I wrote the above mentioned url-anchor library today and I will publish it to Git tomorrow. It currently deepens on my fork of your project. If you don't see any value in this pull-request, you can delete it and I will stick to the fork.

> This exposes the features in the normalize-url library that you accepted earlier today.

I'm just asking this as I get a lot of feature requests and I prefer to have an actual valid reason for adding those than just because it's in a sub-dependency.

But yeah, sounds like a valid use-case, so :+1: 

PR needs rebasing.

sindresorhus commented on index.js:
```diff
@@ -1,9 +1,15 @@
+var arrayUniq    = require('array-uniq'),
+    urlRegex     = require('url-regex'),
+    normalizeUrl = require('normalize-url'),
+    objectAssign = require('object-assign');
```
don't modify unrelated code or change code style

hlysig replied:
Fixed in 09aa8e6f3db2fb23b121053fd423f068d0ab602a

sindresorhus commented:
I think it would be easier to just pass all the option through to normalize-url and refer to it's documentation.

hlysig commented:
Thanks for taking the time reading through the pull-request. I think i've addressed all the points you added in 09aa8e6f3db2fb23b121053fd423f068d0ab602a. If you are happy with these changes you can merge :)

sindresorhus commented on test.js:
```diff
@@ -3,15 +3,64 @@ var assert = require('assert');
+    var expected = ['http://foobar.com/document.html'];
+    var actual = getUrls(text);
+    assert.deepEqual(actual, expected);
+  });
```
These tests are mostly moot, but since you already did them I guess we can just keep them.

hlysig replied:
I think they are relevant where they actually if the parameter is passed down to normalize-url. But if you want I can open up a new pr and remove them.

sindresorhus commented:
Thanks. Available in v5.0.0.

hlysig commented:
No, thank you :) :+1:
